src/ray/protobuf/public/events_node_execution_event.proto

```diff
@@ -0,0 +1,53 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package ray.rpc;
+
+// Message containing the execution information of a node, as observed via GCS.
+// It can be used to capture the full state transition history.
+//
+// For static information associated with this event, see NodeDefinitionEvent.
+message NodeExecutionEvent {
+  enum State {
+    ALIVE = 0;
+    DEAD = 1;
+  }
+
+  message DeathInfo {
+    enum Reason {
+      UNSPECIFIED = 0;
+      EXPECTED_TERMINATION = 1;
+      UNEXPECTED_TERMINATION = 2;
+      AUTOSCALER_DRAIN_PREEMPTED = 3;
+      AUTOSCALER_DRAIN_IDLE = 4;
+    }
+    Reason reason = 1;
+    string reason_message = 2;
+  }
+
+  message StateTimestamp {
+    State state = 1;
+    // Resources (cpu, gpu, etc.) and their counts.
+    map<string, double> resources = 2;
+    DeathInfo death_info = 4;
+    google.protobuf.Timestamp timestamp = 5;
```

Comment:
nit: have state and timestamp next to each other since these are the required fields.